src/main/java/com/friendlyblood/api/domain/models/User.java

```diff
@@ -42,7 +44,7 @@ public User(UserRequestDTO payload){
         this.email = payload.email();
         this.password = payload.password();
         this.name = payload.name();
-        this.bloodType = payload.bloodType();
+        this.bloodType = BloodType.fromString(payload.bloodType());
```

Comment:
Uma vez que você alterasse o enum, como comentei no enum, aqui poderia voltar a ser somente

this.bloodType = payload.bloodType();